win32ss/gdi/ntgdi/freetype.c

```diff
@@ -2994,39 +2988,102 @@ static unsigned int get_bezier_glyph_outline(FT_Outline *outline, unsigned int b
     return needed;
 }
 
-static INT
-IntRequestFontSize(PDC dc, PFONTGDI FontGDI, LONG Width, LONG Height)
+static FT_Error
+IntRequestFontSize(PDC dc, PFONTGDI FontGDI, LONG lfWidth, LONG lfHeight)
 {
+    FT_Error error;
     FT_Size_RequestRec  req;
     FT_Face face = FontGDI->SharedFace->Face;
+    TT_OS2 *pOS2;
+    TT_HoriHeader *pHori;
+    FT_WinFNT_HeaderRec WinFNT;
+    LONG Ascent, Descent, Sum, EmHeight64;
 
-    if (Width < 0)
-        Width = -Width;
+    if (lfWidth < 0)
+    {
+        lfWidth = -lfWidth;
+    }
+    if (lfHeight == 0)
+    {
+        lfHeight = lfWidth;
+    }
+    if (lfHeight == 0)
+    {
+        lfHeight = 13;
+    }
+
+    pOS2 = (TT_OS2 *)FT_Get_Sfnt_Table(face, FT_SFNT_OS2);
+    pHori = (TT_HoriHeader *)FT_Get_Sfnt_Table(face, FT_SFNT_HHEA);
 
-    if (Height < 0)
+    if (!pOS2 || !pHori)
     {
-        Height = -Height;
+        error = FT_Get_WinFNT_Header(face, &WinFNT);
+        if (error)
+            return error;
+
+        FontGDI->tmHeight           = WinFNT.pixel_height;
+        FontGDI->tmAscent           = WinFNT.ascent;
+        FontGDI->tmDescent          = FontGDI->tmHeight - FontGDI->tmAscent;
+        FontGDI->tmInternalLeading  = WinFNT.internal_leading;
+        FontGDI->EmHeight           = FontGDI->tmAscent - FontGDI->tmInternalLeading;
+        if (FontGDI->EmHeight <= 0)
+            FontGDI->EmHeight = 1;
+        if (FontGDI->EmHeight > USHORT_MAX)
+            FontGDI->EmHeight = USHORT_MAX;
```

Comment:
`FontGDI->EmHeight = min(USHORT_MAX, FontGDI->EmHeight);`

Reply:
Both of these can be summarized as:
`FontGDI->EmHeight = min(max(FontGDI->EmHeight, 1), USHORT_MAX);`

Reply:
Yes but it's less readable than keeping it on 2 lines.

Reply:
OK.